core/src/main/java/org/p2p/core/utils/ViewScope.kt

```diff
@@ -0,0 +1,33 @@
+package org.p2p.core.utils
+
+import android.view.View
+import org.p2p.core.R
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+
+val View.viewScope: CoroutineScope
+    get() {
+        val storedScope = getTag(R.id.view_coroutine_scope_tag_id) as? CoroutineScope
+        if (storedScope != null) return storedScope
+
+        val newScope = ViewCoroutineScope()
+        if (isAttachedToWindow) {
+            addOnAttachStateChangeListener(newScope)
+            setTag(R.id.view_coroutine_scope_tag_id, newScope)
+        } else newScope.cancel()
+
+        return newScope
+    }
+
+private class ViewCoroutineScope : CoroutineScope, View.OnAttachStateChangeListener {
+    override val coroutineContext = SupervisorJob() + Dispatchers.Main
+
+    override fun onViewAttachedToWindow(view: View) = Unit
+
+    override fun onViewDetachedFromWindow(view: View) {
+        coroutineContext.cancel()
+        view.setTag(R.id.view_coroutine_scope_tag_id, null)
+    }
+}
```

Comment:
Where do you use it?

Reply:
https://github.com/p2p-org/key-app-android/pull/1242/files/0cee19e3d66c9e113bde2eecf64359e75928ff20#diff-859b1adba07eca3f759298422afe59a6ea9e82618a9f731a2374d66b7b143175R22

Reply:
```
return storedScope ?: ViewCoroutineScope().apply {
            if(isAttachedToWindow) {
                addOnAttachStateChangeListener(this)
                setTag(R.id.view_coroutine_scope_tag_id, this)
            } else {
                cancel()
            }
        }
```